Docs: Update docs to make panel embedding more clear (#97587)
grafana/grafana: docs/sources/dashboards/share-dashboards-panels/_index.md

You must have an authorized viewer permission to see an image rendered by a direct link.

> The same permission is also required to view embedded links unless you have anonymous access permission enabled for your Grafana instance. Please note this is only supported for Grafana Enterprise and Grafana OSS. 

Author: LGTM 👍🏻

> The same permission is also required to view embedded links unless you have anonymous access permission enabled for your Grafana instance. Please note this is only supported for Grafana Enterprise and Grafana OSS. 

Reviewer: ```suggestion
The same permission is also required to view embedded links unless you have anonymous access permission enabled for your Grafana instance.
```

{{< admonition type="note" >}}

> Anonymous access permission is not available in Grafana Cloud.

Reviewer: ```suggestion
Anonymous access permission is not available in Grafana Cloud. This feature is only supported for Grafana Enterprise and Grafana Open Source.
```

### Share an embed

> You can share a panel by embedding it on another website using an iframe. Users must be signed into Grafana to view the panel even if it is from a Public Dashboard in Grafana Cloud. 

Reviewer: The combination of this update and the one at line 301 is confusing to me. Once we've clarified line 301, then I'll come back to this. But it seems like what you've added here is saying the same thing as what's in the note on line 273, no?

Author: You're not wrong, it's redundant. I'm just trying to make it apparent to folks that public vs private won't matter, they've still gotta be signed in due to `X-Frame-Options` to prevent displaying.

> You can share a panel by embedding it on another website using an iframe. Users must be signed into Grafana to view the panel even if it is from a Public Dashboard in Grafana Cloud. 

Reviewer: ```suggestion
You can share a panel by embedding it on another website using an iframe. Users must be signed into Grafana to view the panel unless you have anonymous access permission enabled for your Grafana instance.
```

{{< admonition type="note" >}}

> Anonymous access permission is no longer available for Grafana Cloud. 

Reviewer: ```suggestion
Anonymous access permission is no longer available for Grafana Cloud. This feature is only supported for Grafana Enterprise and Grafana Open Source.
```

></iframe>
```

> The result is an interactive Grafana visualization embedded in an iframe. This is only supported on Grafana OSS. You cannot embed a private or public panel within Grafana Cloud. 

Reviewer: What exactly do you mean isn't supported here? That you can't embed the dashboard _back into_ your Cloud instance and that you can with OSS (and I'm assuming Enterprise)?

Author: Just to clarify that even if a panel in a dashboard is publicly accessible if you try to embed it within an iframe it still won't display without the `allow_embedding` which support largely will not request. The goal is to reduce contact points, make it apparent public vs private doesn't matter.

Reviewer: Just to clarify, do you mean it won't display for Cloud users unless they're signed in or _not at all_? If the first, then this bit doesn't help here because it makes it sound like you can't use the feature at all if you're on Cloud.

```suggestion
The result is an interactive Grafana visualization embedded in an iframe.
```

Author: Sorry for the confusion! It won't display unless they're signed in which sort of defeats the purpose of embedding a dashboard elsewhere. I like your language better, LGTM! Ty!

Reviewer: Understood! That's probably a larger question to bring up with the sharing squad of whether this feature really make sense for Cloud users. It sounds like this is a common issue you're coming up against; do you want me to raise it with the squad?
